components/collection/HeroButtonEditCollection.vue

```diff
@@ -67,14 +73,25 @@ const editCollection = async (collection: UpdateCollection) => {
 
   await transaction({
     interaction: Collections.UPDATE_COLLECTION,
-    collectionId: route.params.id.toString(),
+    collectionId: collectionId,
     collection,
     update: {
       metadata: updateMetadata(collection, collectionMetadata.value),
       max: collection.max !== collectionMetadata.value.max,
+      permission: shouldUpdatePermission(collection.mintingSettings, collectionPermissionSettings.value!),
     },
     urlPrefix: urlPrefix.value,
     successMessage: $i18n.t('edit.collection.success'),
   })
 }
+
+watchEffect(async () => {
+  const { apiInstance } = useApi()
+  const api = await apiInstance.value
+  const config = await api.query.nfts.collectionConfigOf(collectionId)
+  const mintSettings = (config.toHuman() as { mintSettings: CollectionMintSetting }).mintSettings
+
+  mintSettings.price = mintSettings.price?.replaceAll(',', '')
+  collectionPermissionSettings.value = mintSettings
+})
```

Comment:
since this was added lets disable the `edit collection` button if `!collectionMetadata`

Reply:
added